src/Tribe/Repositories/Post_Repository.php

```diff
@@ -0,0 +1,245 @@
+	 *
+	 * @var string
+	 */
+	protected $filter_name = 'tickets_post';
+
+	/**
+	 * Post_Repository constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->default_args = array_merge( $this->default_args, [
+			'orderby'                      => 'date',
+			'post_status'                  => 'any',
+			'post_type'                    => 'any',
+			// We'll be handling the dates, let's mark the query as a non-filtered one.
+			'tribe_suppress_query_filters' => true,
+		] );
+
+		$this->schema = array_merge( $this->schema, [
+			// These filter methods are added by the Event trait.
+			'ends_after'           => [ $this, 'filter_by_ends_after' ],
+
+			// These filter methods are added by the Post_Tickets trait.
+			'cost'                 => [ $this, 'filter_by_cost' ],
+			'cost_currency_symbol' => [ $this, 'filter_by_cost_currency_symbol' ],
+			'has_tickets'          => [ $this, 'filter_by_has_tickets' ],
+			'has_rsvp'             => [ $this, 'filter_by_has_rsvp' ],
+
+			// These filter methods are added by the Post_Attendees trait.
+			'has_attendees'        => [ $this, 'filter_by_has_attendees' ],
+			'attendee'             => [ $this, 'filter_by_attendee' ],
+			'attendee__not_in'     => [ $this, 'filter_by_attendee_not_in' ],
+			'attendee_user'        => [ $this, 'filter_by_attendee_user' ],
+
+			// This is not yet working, it needs more debugging to determine why it's not functional yet.
+			//'attendee_user__not_in' => [ $this, 'filter_by_attendee_user_not_in' ],
+		] );
+	}
+
+	/**
+	 * Returns an array of the attendee types handled by this repository.
+	 *
+	 * Extending repository classes should override this to add more attendee types.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function attendee_types() {
+		return [
+			'rsvp'           => 'tribe_rsvp_attendees',
+			'tribe-commerce' => 'tribe_tpp_attendees',
+		];
+	}
+
+	/**
+	 * Returns the list of meta keys relating an Attendee to a Post (Event).
+	 *
+	 * Extending repository classes should override this to add more keys.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function attendee_to_event_keys() {
+		return [
+			'rsvp'           => '_tribe_rsvp_event',
+			'tribe-commerce' => '_tribe_tpp_event',
+		];
+	}
+
+	/**
+	 * Returns the meta key relating an Attendee to a User.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function attendee_to_user_key() {
+		return '_tribe_tickets_attendee_user_id';
+	}
+
+	/**
+	 * Handles the `order_by` clauses for events
+	 *
+	 * @since 4.9.7
+	 *
+	 * @param string $order_by The key used to order events; e.g. `event_date` to order events by start date.
+	 */
+	public function handle_order_by( $order_by ) {
+		$check_orderby = $order_by;
+
+		if ( ! is_array( $check_orderby ) ) {
+			$check_orderby = explode( ' ', $check_orderby );
+		}
+
+		$timestamp_key = 'TIMESTAMP(mt1.meta_value)';
```

Comment:
Can we be sure that `mt1` will always be the timestamp?

Reply:
I copied this over from TEC, I'm not sure yet. We're not using this specific functionality just yet though. I am working on getting more testing coverage in the upcoming sprint for this.

Reply:
Actually this is getting used right now https://github.com/moderntribe/event-tickets/pull/1659/files#diff-fda7a1bbb895152d4e2aa1ec12a7d2aeR98-R99 and it seems to be using mt1. I'll have to get into the internals and see if we can abstract it any further for cases where it's not that alias, but that same issue exists in TEC so it might be better to tackle it there in TEC first.